docs-v2/guides/getting-started/authorize-an-api-from-your-app.mdx

```diff
@@ -43,79 +70,63 @@ api.post('/sessionToken', (req, res) => {
 
-```js
-import Nango from '@nangohq/frontend';
-
-const nango = new Nango();
-const connect = nango.openConnectUI();
+  Required fields:
+  - `end_user.id`: Your internal ID of the user who initiated the authorization flow.
+  - `end_user.email`: The email of the user who initiated the authorization flow.
 
-// Call the endpoint created in Step 1
-const res = await fetch('/sessionToken', { method: 'POST' });
-connect.setSessionToken(res.sessionToken);
-```
+  While some connections may belong to an organization rather than a specific user, in most cases, a user initiates the authorization flow. In such scenarios:
+  - You should store the user’s ID and email along with the relevant organization details. Sometimes, the user email in your app will not match the email they use to connect their external account, which is fine. 
+  - If you don’t have a user ID or email, you can use placeholders. This won’t affect how Nango operates.
 
-<Tip>It's recommended to set the `sessionToken` asynchronously to be able to display the UI loading to your users before it's ready for better UX</Tip>
+  <Info>
+    The user’s email is used in the Nango UI to retrieve & display their company’s logo, improving visual clarity.
+  </Info>
+</Accordion>
 
-## Step 3: Link Connection ID to your user
+## Trigger the auth flow (frontend)
 
-In **your frontend**, the Connect UI will send back an event when a user connects or closes the modal. You can register an event listener and respond appropriately.
+In **your frontend**, load the Nango frontend SDK, retrieve the session token from the backend, and trigger the authorization flow. 
 
-When a user completes a flow, you will receive a `connect` event. This event contains the `providerConfigKey`, which is the id of your integration and a `connectionId`, which is the auto-generated id that represents the couple user + integration.
+### Option 1: Use the Nango Connect UI
 
 ```js
-[...]
-
-function saveConnectionId(authResults) {
-  await fetch('/connection', { method: 'POST', body: authResults });
-}
-
-nango.openConnectUI({
-  sessionToken: await getSessionToken(),
+import Nango from '@nangohq/frontend';
 
-  // Listen to events
+const nango = new Nango();
+const connect = nango.openConnectUI({ 
   onEvent: (event) => {
-    if (event.type === 'connect') {
-      void saveConnectionId(event.payload);
+    if (event.type === 'close') {
+      // Handle modal closed.
+    } else if (event.type === 'connect') {
+      // Handle auth flow successful.
     }
   },
 });
-```
-
-In your backend, associate this connectionId to your end user.
 
-```javascript
-api.post('/connection', (req, res) => {
-  await User.update({
-    connectionId: req.body.connectionId,
-    integrationId: req.body.providerConfigKey,
-  });
-
-  res.status(200).send({
-    success: true
-  });
-});
```

Comment:
The reason I added this code snippet is that it's impossible to use webhook right now with a session token, you'll receive a random connectionId with no link to anything

Reply:
This should be fix thanks to your change, and the docs are updated.